### ISSUE TITLE
Slack channel stuck on loading screen

### ISSUE THREAD

**Issue by alexdevero** [meetfranz/franz#366]
### Expected Behavior
Slack channel is loaded properly and content is displayed

### Current Behavior
Slack channel is sometimes stuck on loading screen in an infinite loop.

### Screenshots (if appropriate):
![slack-channel](https://user-images.githubusercontent.com/4237724/33375948-0ceb09a2-d50c-11e7-8581-3011f5452ff8.PNG)

### Console log (service dev tools):
```
rollup-core_required_libs.8e73dad….min.js:93 Warning: a promise was created in a handler but was not returned from it
    at M (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:17087)
    at O.r._warn (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:13380)
    at Object.I [as checkForgottenReturns] (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:16899)
    at O._settlePromiseFromHandler (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:50842)
    at O._settlePromise (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:51575)
    at O._settlePromiseCtx (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:52137)
    at u._drainQueue (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:2946)
    at u._drainQueues (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:2999)
    at drainQueues (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:1264)
    at MutationObserver.<anonymous> (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:71779)
From previous event:
    at O.P [as _captureStackTrace] (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:16419)
    at O._resolveFromExecutor (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:50178)
    at new O (https://a.slack-edge.com/bv1-1/rollup-core_required_libs.8e73dada0d381d3f8d4b.min.js:93:43920)
    at C:\Users\Alex\AppData\Local\Programs\franz\resources\app.asar\node_modules\electron-spellchecker\lib\spell-check-handler.js:73:99
    at SpellCheckHandler.switchLanguage (C:\Users\Alex\AppData\Local\Programs\franz\resources\app.asar\node_modules\electron-spellchecker\lib\spell-check-handler.js:465:7)
    at Spellchecker.enable (C:\Users\Alex\AppData\Local\Programs\franz\resources\app.asar\webview\spellchecker.js:22:25)
    at EventEmitter._electron.ipcRenderer.on (C:\Users\Alex\AppData\Local\Programs\franz\resources\app.asar\webview\plugin.js:39:20)
    at emitTwo (events.js:106:13)
    at EventEmitter.emit (events.js:194:7)
v @ rollup-core_required_libs.8e73dad….min.js:93
rollup-core_required_libs.8e73dad….min.js:93 Uncaught Error: cannot enable cancellation after promises are in use
    at Function.r.config (rollup-core_required_libs.8e73dad….min.js:93)
    at e (rollup-core_required_ts.1dbcd33….min.js:1)
    at Object.e [as boot] (rollup-core_required_ts.1dbcd33….min.js:1)
    at messages:2270
rollup-core_required_libs.8e73dad….min.js:201 SoundManager 2: No Flash detected. Trying HTML5-only mode.
rollup-core_required_libs.8e73dad….min.js:201 SoundManager V2.97a.20150601 (HTML5-only mode)
rollup-core_required_libs.8e73dad….min.js:201 SoundManager 2 HTML5 support: mp3 = true, mp4 = true, ogg = true, opus = true, wav = true
rollup-core_required_libs.8e73dad….min.js:201 SoundManager 2: Ready. ✓
webview.js:5 getTeamIcon 0
webview.js:10 Uncaught TypeError: Failed to execute 'getComputedStyle' on 'Window': parameter 1 is not of type 'Element'.
    at getTeamIcon (webview.js:10)
    at setTimeout (webview.js:38)
```

### Steps to Reproduce (for bugs)

1. Add Slack channel.
2. Deactivate it.
3. Activate it.
4. Open it.

### Your Environment

* Franz Version used: 5.0.0-beta.14
* Operating System and version: Win 10 (1709)

**Comment by miscbits** [comment]
Im getting this same behavior

**Comment by alexdevero** [comment]
@miscbits For now, way to "fix" it is by disabling and enabling the channel again.

**Comment by miscbits** [comment]
Thanks. Worth noting, I'm on Ubuntu 16.04

**Comment by adlk** [comment]
Can you guys do me a favor and run the following code in the _Service Dev Tools_ once Slack is fully loaded?

```
document.querySelector('#team_menu').click();
const icon = document.querySelector('.team_icon');

console.log(icon);
```

**Comment by CCRDude** [comment]
Here's my output once Slack is fully loaded:
```
document.querySelector('#team_menu').click();
undefined
const icon = document.querySelector('.team_icon');
undefined
console.log(icon);
VM1010:1 null
undefined
```

**Comment by adlk** [comment]
@CCRDude are you on the Slack version with "all unreads" in the Menu?

**Comment by colinodell** [comment]
Franz just prompted me to "reload services".  After doing so I am now experiencing this same issue.  Ubuntu 16.04 and Franz 5.0.0-beta.14.

To replicate the issue, I killed Franz and started it back up.  I immediately opened the service dev tools as Slack began loading in and saw this output:

![image](https://user-images.githubusercontent.com/202034/33397791-bf1f889c-d51a-11e7-9617-062e22b681cd.png)

I then cleared the dev tools console and clicked Reload Service - this caused Slack to load in just fine with the following console output:

![image](https://user-images.githubusercontent.com/202034/33397876-0bb6dc6e-d51b-11e7-874f-c023d3d883c8.png)
